notanorm/ddl_helper.py

```diff
@@ -0,0 +1,143 @@
+from collections import defaultdict
+from typing import Tuple, Dict, List
+
+import sqlglot
+from sqlglot import parse, exp
+
+from .model import DbType, DbCol, DbIndex, DbTable, DbModel, ExplicitNone
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class DDLHelper:
+    TYPE_MAP = {
+        exp.DataType.Type.INT: DbType.INTEGER,
+        exp.DataType.Type.BIGINT: DbType.INTEGER,
+        exp.DataType.Type.BOOLEAN: DbType.BOOLEAN,
+        exp.DataType.Type.BINARY: DbType.BLOB,
+        exp.DataType.Type.VARCHAR: DbType.TEXT,
+        exp.DataType.Type.CHAR: DbType.TEXT,
+        exp.DataType.Type.TEXT: DbType.TEXT,
+        exp.DataType.Type.VARIANT: DbType.ANY,
+        exp.DataType.Type.DECIMAL: DbType.DOUBLE,
+        exp.DataType.Type.DOUBLE: DbType.DOUBLE,
+        exp.DataType.Type.FLOAT: DbType.FLOAT,
+    }
+
+    FIXED_MAP = {
+        exp.DataType.Type.CHAR,
+        #  todo: add support for varbinary vs binary in sqlglot
+        #        exp.DataType.Type.BINARY
+    }
+
+    def __init__(self, ddl, *dialects):
+        if not dialects:
+            dialects = ("mysql", "sqlite")
+
+        last_x = None
+
+        for dialect in dialects:
+            if dialect == "mysql":
+                # bug sqlglot doesn't support varbinary
+                ddl = ddl.replace("varbinary", "binary")
+            try:
+                res = parse(ddl, read=dialect)
+                self.ddl = res
+                break
+            except sqlglot.ParseError as ex:
+                last_x = ex
+
+        if last_x:
+            raise last_x
+
+    def __columns(self, ent) -> Tuple[Tuple[DbCol, ...], DbIndex]:
```

Comment:
docstrings on these helpers would be helpful, not self-descriptive

Reply:
cool